### DISCUSSION TITLE
Changing status from single chart to multiple charts.

### DISCUSSION THREAD

**Discussion by nikobearrr** [netdata/netdata#14880] [Q&A] [answered]
I use netdata with go.d to monitor ES cluster.
I just updated from v1.37.1 to v1.38.1 and found our the cluster health metric has changed.

Before it used to be `elasticsearch_node.status` and each of the status (green, yellow, red) would be present in a single chart.
Now, each status is getting it's own chart - one for green, one for yellow, one for red.

My question/issue is that when looking into the netdata dashboard it is easier to see when I have all status in a single chart. Now I need to look into 3 of them. Same goes when I export to grafana. When I use a single chart/metric, I can set green=0, yellow=1 and red=2. However, when there are 3 I cannot visualize it so easy.

So to formulate a question: is this the way netdata plans to do with all [status] metrics? Is there a way for me to combine multiple ones in a single chart?

**Comment by ilyam8** [comment, marked as answer]
Hi, @nikobearrr. 

> Now, each status is getting it's own chart - one for green, one for yellow, one for red.

A small correction: it is not a chart per status but a dimension per status ([one chart](https://github.com/netdata/go.d.plugin/blob/f255d2c73b69b9666e1a5c302c2748fc182ca67a/modules/elasticsearch/charts.go#L548-L560)).

This is needed because aggregation over time and over instances is completely broken when we encode the status as int (green=0, yellow=1 and red=2): any aggregation (min, max, avg, etc.) method produces an unusable value. The is no such problem if we have a dimension per status (e.g. aggregation (sum) of several `elasticsearch.cluster_health_status` charts is nice because it gives you the number of clusters in every status).

>  is this the way netdata plans to do with all [status] metrics?

Yes, with all state/status metrics because that is the only valid way to have not broken aggregation. It is similar to OpenMetrics [StateSet](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#stateset).

**Comment by nikobearrr** [comment]
Thank you for the thorough answer @ilyam8.

For my use case I export the metrics into graphite and use a max() when dealing with aggregation based on time. 
I can still do alerts as I expect them (alert when yellow/red > 0), however the graph I present now has 3 extra dimension. My goal is to see the `status` over time. And by having it as an `int` it is very easy as you see a stair case pattern. Having them as booleans does not make it very easy to see, at least not in my particular usage.

If it will not be hard and will not lead to too much code duplication I would like to post it as a feature request - have an option for such charts to be 1 vs 3 dimensions. I tried to find similar cases in the issues/discussions, but I failed to do so. I do not know how popular such demand would be.

**Reply by ilyam8** [reply]
@nikobearrr No, there are no plans to change it back (or even make it an option) because the way it was is broken.
